definitions/npm/redux-form_v5.x.x/flow_v0.53.x-/redux-form_v5.x.x.js

```diff
@@ -0,0 +1,168 @@
+
+
+ */
+
+declare module 'redux-form' {
+  declare export type InputProps = {
+    name: string,
+    value: string | boolean,
+    valid: boolean,
+    invalid: boolean,
+    dirty: boolean,
+    pristine: boolean,
+    active: boolean,
+    touched: boolean,
+    visited: boolean,
+    autofilled: boolean,
+    error?: string,
+    onChange: (eventOrValue: SyntheticEvent<any> | string | boolean) => mixed,
+    onUpdate: (eventOrValue: SyntheticEvent<any>  | string | boolean) => mixed,
+    onBlur: (eventOrValue: SyntheticEvent<any>  | string | boolean) => mixed,
+    onDragStart: Function,
+    onDrop: Function,
+    onFocus: Function,
+  };
+
+  declare export type MetaProps = {
+    active: boolean,
+    autofilled: boolean,
+    asyncValidating: boolean,
+    dirty: boolean,
+    dispatch: Function,
+    error?: string,
+    invalid: boolean,
+    pristine: boolean,
+    submitting: boolean,
+    touched: boolean,
+    valid: boolean,
+    visited: boolean,
+  };
+
+  declare export type FieldInputProps = {
+    input: InputProps,
+    meta: MetaProps,
+  };
+
+  declare type FunctionComponent<P> = (props?: P) => ?React$Element<any>;
+  declare type ClassComponent<P, S> = Class<React$Component<P, S>>;
+  declare type Component<P> = FunctionComponent<P> | ClassComponent<P, any>;
+
+  declare export type FieldProps<P> = {
+    name: string,
+    placeholder?: ?string,
+    label?: string,
+    value?: ?(string | boolean),
+    component: ClassComponent<P, void> | FunctionComponent<P> | string
+  } & $Diff<P, FieldInputProps>;
+
+  declare export type RegisteredField<T> = {
+    name: $Keys<T>,
+    type: string,
+  };
+
+  declare export type FormState<T> = {
+    values: T,
+    initial: T,
+    registeredFields: Array<RegisteredField<T>>,
+  };
+
+  declare export type FormProps = {
+    active: string,
+    asyncValidate: (values: Object, dispatch: Function, props: Object) => Promise<void>,
+    asyncValidating: string | boolean,
+    destroyForm: Function,
+    dirty: boolean,
+    error: string,
+    fields: { [fieldName: string]: InputProps },
+    handleSubmit: (data: { [field: string]: string }) => void | Promise<any>,
+    initializeForm: (data:Object) => any,
+    invalid: boolean,
+    pristine: boolean,
+    resetForm: Function,
+    formKey: string,
+    submitting: boolean,
+    submitFailed: boolean,
+    touch: (...fields: Array<string>) => void,
+    touchAll: () => void,
+    untouch: (...fields: Array<string>) => void,
+    untouchAll: () => void,
+    valid: boolean,
+    values: Object
+  };
+
+  declare export type FormConfig = {
+    fields: Array<string>,
+    form: string,
+    alwaysAsyncValidate?: boolean,
+    asyncBlurFields?: Array<string>,
+    asyncValidate?: (values: Object, dispatch: Function, props: Object) => Promise<void>,
+    destroyOnUnmount?: boolean,
+    formKey?: string,
+    getFormState?: (state: Object, reduxMountPoint: string) => mixed,
+    initialValues?: { [field: string]: string },
```

Comment:
initialValues can be also nested object, at least thats how we use it in our project.